src/serac/numerics/functional/functional.hpp

```diff
@@ -301,7 +301,7 @@ class Functional<test(trials...), exec> {
 
   /// @overload
   template <int dim, int... args, typename lambda, typename qpt_data_type = Nothing>
-  void AddDomainIntegral(Dimension<dim>, DependsOn<args...>, const lambda& integrand, Domain& domain,
+  void AddDomainIntegral(Dimension<dim>, DependsOn<args...>, const lambda& integrand, const Domain& domain,
```

Comment:
the DG PR requires making changes to the domain class when defining integrals, so we might have to revert this back to non-const `Domain & domain` soon (or use `mutable` stuff)

Reply:
Noted. There's the usual virality: If we take away the `const` here, it will have to propagate down to `Functional` (which is why I added it here). I'll let you pick which.